src/components/SpacemeshApiClientMock/index.tsx

```diff
@@ -0,0 +1,231 @@
+import * as React from 'react';
+import { ISpacemeshApi } from '../../model';
+import { ITransaction, TransactionStatus, MockData} from 'src/types';
+
+interface IProps extends ISpacemeshApi{
+    apiReady: boolean;
+}
+
+interface IState {
+    connectedToApi: boolean;
+}
+
+const SpacemeshApiClientMock = <P extends IProps>(Component: React.ComponentType<P>) => {
+    return class SpacemeshApiClient extends React.Component<P & IProps, IState> implements ISpacemeshApi {
+        constructor(props: P & IProps) {
+            super(props);
+            this.state = {
+                connectedToApi: false,
+            }
+        }
+
+        public componentDidMount() {
+            this.connectSpacemeshApiClient();
+        }
+
+        public render() {
+
+            return <Component
+            {...this.props} 
+            apiReady = {this.state.connectedToApi}
+            boradCastTransaction = {this.boradCastTransaction}
+            getAddressBalance = {this.getAddressBalance}
+            getTransactionFromAddress = {this.getTransactionFromAddress}
+            buySMCFromExchange = {this.buySMCFromExchange}
+            getCurrentSMCPrice = {this.getCurrentSMCPrice}
+            getSMCfromTestNetTap = {this.getSMCfromTestNetTap}
+            getTransactionStatus = {this.getTransactionStatus}
+            />;
+        }
+
+        /**
+         * Broadcast a transaction to the Spacemesh network
+         * @param tx raw transaction to broadcast
+         * @returns transactions hash
+         */
+        public boradCastTransaction = (tx: ITransaction): Promise<MockData.IBroadCastTransactionResponse> => {
+            return new Promise(resolve => {
+                setTimeout(() => {
+                    resolve({
+                        id:1,
+                        jsonrpc: '1.0',
+                        params: [tx],
+                        result: '0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331'
```

Comment:
Using a constant here make the mock too rigid for my taste.  How about letting the user set the next result?